text/0089-scoped-api-keys.md

```diff
@@ -0,0 +1,128 @@
+- Title: Scoped API Keys
+- Start Date: 2021-10-15
+- Specification PR: [#89](https://github.com/meilisearch/specifications/pull/89)
+- Discovery Issue: [#51](https://github.com/meilisearch/product/issues/51)
+
+# Scoped API Keys
+
+## 1. Functional Specification
+
+### I. Summary
+
+The SDKs can generate `Scoped API Keys` inheriting from a MeiliSearch's `API key` generated to force filters during a search for an end-user. A `Scoped API Key` is generated on the user server-side code to be used by an end-user. It allows users to have multi-tenant indexes and thus restricts access to documents depending on the end-user making the search request.
+
+### II. Motivation
+
+`Scoped API Keys` are introduced to solve multi-tenant use-case. By managing this, we reduce one of the last major deal-breakers that makes users not choose MeiliSearch as a solution despite all our advantages.
+
+Users regularly request Multi-Tenant indexes. Users today need to set up workarounds to meet this need. Some of them implement reverse-proxy or managed authentication systems like Hasura or Kong to filter what can and cannot be read. Others decide to use server code as a facade to implement the access restriction logic before requesting MeiliSearch. It is difficult to maintain, less efficient, and requires necessary skills that not everyone has.
+
+### III. Glossary
+
+| Term               | Definition |
+|--------------------|------------|
+| Main Key         | This is the main key that allows the creation of other API keys. The main key is defined by the user when launching MeiliSearch. |
+| API Key            | API keys are stored and managed from the endpoint `/keys` by the main key holder. These are the keys used by the technical teams to interact with MeiliSearch at the level of the client code. |
```

Comment:
>  to interact with MeiliSearch at the level of the client code.

What do you mean with this?

Reply:
Client code here means anything that can communicate and interact with MeiliSearch. A curl script, an SDK etc